### ISSUE TITLE
Training a multi-branch CNN using a generator causes a global loss ambiguious evolution

### ISSUE THREAD

**Issue by Emnaa12** [keras-team/keras#14016]
I ' am trying to train a network with two separate branches, each branch is a ResNet50. That means I have two partial losses and a global loss which is a weigthed summation of partial ones. I used a generator to get batches. During the training, the global loss is much higher than the theorical weighted summation of the obtained partial ones. Here is the code:

     import numpy as np
     from keras_preprocessing.image import ImageDataGenerator
     from keras.models import Model 
     import keras
     import resnet

     img_channels=3
     img_rows=224
     img_cols=224
     num_classes=10
     image_dir1='Path_to_directory1'
     image_dir2='Path_to_directory2'

     def multi_loss_model():
         model_rgb  = resnet.ResnetBuilder.build_resnet_50((img_channels, img_rows,                        img_cols), num_classes)
        model_disp = resnet.ResnetBuilder.build_resnet_50((img_channels, img_rows, img_cols), num_classes)
        custom_resnet_model = Model(inputs=[model_rgb.input,model_disp.input],         outputs=[model_rgb.output,model_disp.output])

        return custom_resnet_model

    model_resnet_multi_class= multi_loss_model()

    model_resnet_multi_class.compile(loss='categorical_crossentropy', 
                  loss_weights=[0.5,0.5],
                  optimizer='sgd', metrics=['accuracy'])

      batch_size=32
      image_size=(224,224)
      train_samples=340
      validation_samples=84
      epoch_num=100
      train_datagen = ImageDataGenerator(rescale=1./255, validation_split=0.2) 

      def multiple_outputs(generator, image_dir1, image_dir2,batch_size, image_size,             subset):
          gen1 = generator.flow_from_directory(
                     image_dir1,
                    target_size=image_size,
                    batch_size=batch_size,
                    class_mode='categorical', shuffle=True, seed=42,
                    subset=subset)
          gen2 = generator.flow_from_directory(
                    image_dir2,
                    target_size=image_size,
                    batch_size=batch_size,
                    class_mode='categorical',shuffle=True, seed=42,
                    subset=subset)
    
          while True:
              gnext1 = gen1.next()
              gnext2 = gen2.next()
              yield [gnext1[0],gnext2[0]], [gnext1[1],gnext2[1]]
        
      train_generator = multiple_outputs(
                train_datagen,
                image_dir1=image_dir1,
                image_dir2=image_dir2,
                batch_size=batch_size,
                image_size=image_size,
                subset='training')
     
      validation_generator = multiple_outputs(
          train_datagen,
          image_dir1=image_dir1,
          image_dir2=image_dir2,
          batch_size=batch_size,
          image_size=image_size,
          subset='validation')

      model_resnet_multi_class.fit_generator(train_generator,
                          steps_per_epoch = train_samples // batch_size,
                          validation_data = validation_generator, 
                          validation_steps = validation_samples // batch_size, 
                          epochs = epoch_num)

Here is the result of the first training epoch

    Epoch 1/10   
    10/10 [==============================] - 512s 51s/step - loss: 
    12.7830 - dense_5_loss: 2.0693 - dense_6_loss: 2.2488 - dense_5_accuracy: 
    0.3031 - dense_6_accuracy: 0.2406 - val_loss: 21.3956 - val_dense_5_loss: 14.6704 
    - val_dense_6_loss: 6.3336 - val_dense_5_accuracy: 0.1562 -val_dense_6_accuracy: 
    0.0781
Could anyone explain to me what is wrong with my code to result in such global loss aspect? Theoretically, it should be the average of the two partial losses since I affected a weight of 0.5 for each loss. When I tried the same with one branch, the global loss computation works correctly. However, it is not the case with two branches.

**Comment by Emnaa12** [comment]
@fchollet I would be so gratefull if you could help me and figure out the key problem of my issue.

**Comment by Saduf2019** [comment]
@Emnaa12 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
